File: 백준/Silver/1940. 주몽/주몽.cc
#include <bits/stdc++.h>
using namespace std;

int N, M, ret;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    cin >> N >> M;
    vector<int> v(N);
    for (int i=0; i<N; i++) cin >> v[i];

    if (M > 200000) cout << 0 << '\n';
    else {
        int l = 0;
        int r = v.size()-1;
        sort(v.begin(), v.end());

        while (l < r) {
            if (v[l] + v[r] > M) r--;
            else if (v[l] + v[r] == M) ret++, l++;
            else l++;
        }
        cout << ret << '\n';
    }
    return 0;
}

File: 백준/Silver/1940. 주몽/주몽.py
N = int(input())
M = int(input())
arr = list(map(int, input().split()))
arr.sort()

ret = 0
l, r = 0, N-1

if l + r > 200000:
    print(0)
else:
    while l < r:
        if arr[l] + arr[r] < M:
            l += 1
        elif arr[l] + arr[r] == M:
            ret += 1
            r -= 1
        else:
            r -= 1

print(ret)